### ISSUE TITLE
Nested ContentDialog leads crash System.Runtime.InteropServices.COMException

### ISSUE THREAD

**Issue by prabakaran-sangameswaran** [microsoft/microsoft-ui-xaml#6473]
### Describe the bug

I tried to display the one content dialog inside another content dialog (Nested ContentDialog), but when i open the second content dialog it leads to a crash.

### Steps to reproduce the bug

1. Run the [Nested_ContentDialog.zip](https://github.com/microsoft/microsoft-ui-xaml/files/7703527/Nested_ContentDialog.zip) sample and click "Open dialog 1".
2. First content dialog will be displayed.
3. Now, click "Open dialog 2" to display the second content dialog.
4. Crash occurs.

### Expected behavior

Second content dialog should be displayed.

### Screenshots

![image](https://user-images.githubusercontent.com/57438309/145807105-d40e43bd-f4f2-4471-af42-724cdeb8cc32.png)

### NuGet package version

WinUI 3 - Windows App SDK 1.0 (If you're seeing your issue in older previews of WinUI 3, please try this release)

### Windows app type

- [ ] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

October 2020 Update (19042)

**Comment by saiganesh-sakthivel** [comment]
Hi @gabbybilka,

I am also facing the same issue when loading custom Flyout instead of a second content dialog.

Regards,
SaiGanesh Sakthivel

**Comment by andreaslennartz** [comment]
I encounter the same issue when trying to use a another pre-made component which has a ContentDialog built in - this component can't be used inside another ContentDialog.

**Comment by andreaslennartz** [comment]
As far as I can see this is still not resolved with the current release of WinUI (1.3). Any updates on this?

**Comment by jlipford** [comment]
Any update on this?

**Comment by beeradmoore** [comment]
I believe this is [by design](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.contentdialog?view=windows-app-sdk-1.6), 

>
> Only one ContentDialog can be shown at a time. To chain together more than one ContentDialog, handle the [Closing](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.contentdialog.closing?view=windows-app-sdk-1.6#microsoft-ui-xaml-controls-contentdialog-closing) event of the first ContentDialog. In the [Closing](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.contentdialog.closing?view=windows-app-sdk-1.6#microsoft-ui-xaml-controls-contentdialog-closing) event handler, call [ShowAsync](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.contentdialog.showasync?view=windows-app-sdk-1.6#microsoft-ui-xaml-controls-contentdialog-showasync(microsoft-ui-xaml-controls-contentdialogplacement)) on the second dialog to show it.
> 
> There can only be one ContentDialog open per thread at a time. Attempting to open two ContentDialogs will throw an exception, even if they are attempting to open in separate app windows.
>

**Comment by andreaslennartz** [comment]
Hi @beeradmoore ,

thanks for you reply. That actually make sense, thanks for taking the time to reply here. 

Just a little bit of context here: The root cause of this issue is originated from the Syncfusion CalendarDatePicker Control, which opens up a ContentDiaglog inside the control when the user wants to pick a date. This doesn't work when using this Control inside a ContentDialog.  
I am guessing it is a bad idea to use a ContentDialog for the DatePicker in the first place.

**Comment by beeradmoore** [comment]
Makes sense why it's broken there. My program has a fake content dialog which may help fix the issue for you (it may add new issues). I think I originally made it to have more customisation of a ContentDialog. Although this issue is 3 years old and you have worked around it, maybe this will help others.

My use case was I have a list of video games. Clicking one opens a [FakeContentDialog](https://github.com/beeradmoore/dlss-swapper/blob/development/src/UserControls/FakeContentDialog.cs). That displays details about the game. Clicking on from there I have various options for different things. Clicking one of those opens a real ContentDialog with a list in it. Some of the items in the list need to be downloaded before they are used.

My idea was when you click one that needs to be downloaded I preset a second ContentDialog with a progress bar. That was when I found this issue. I tried a fake dialog over a fake dialog but that has issues with both dialogs receiving the same input. My solution there was just to put the progress bar in the list view row.
